### PR TITLE
Update backend to support windows store player IDs

### PR DESCRIPTION
**Pull request by cemathey**
* Update the kill/team kill log parsing regex to support windows store IDs
* Refactor the player info command result parsing for easier testing
* Update various steam utilities to only call the steam API if the player ID is an actual steam ID
* Add various tests

I did build this, run the tests (some of the tests fail but it's unrelated to this and should be failing on master as well), assign a steam API key and connect it to a server with actual players on it and let it run a bit and poked around, I believe this works correctly.

I am curious/suspect there might be a better way to do the filtering decorators and I am all ears.

I do have to fix some cache handling errors from #410 but you'll only see that issue if redis is unavailable, and docker desktop (where I almost exclusively do this stuff) likes to have random DNS lookup issues which is how I noticed.

### CODE REVIEW

File: rcon/rcon.py [hunk 1]
```diff
@@ -44,7 +44,9 @@
     ALL_ROLES,
     ALL_ROLES_KEY_INDEX_MAP,
     INDEFINITE_VIP_DATE,
+    _get_default_info_dict,
```
Comment (FlorianSW) [inline rcon/rcon.py:47]: when we're on that already, some suggestions:
- make this method non-private (remove the underscore)
- rename the method to something more meaningful, e.g.: `new_from_player_name`

what do you think?

File: rcon/rcon.py [hunk 2]
```diff
@@ -44,7 +44,9 @@
...
     get_server_number,
+    parse_raw_player_info,
```
Comment (FlorianSW) [inline rcon/rcon.py:49]: to make it consistent with my suggestion from above, what about renaming this one to:
`new_from_player_info`? :)

File: rcon/rcon.py [hunk 3]
```diff
@@ -105,9 +107,8 @@ class Rcon(ServerCtl):
     player_info_regexp = re.compile(r"(.*)\(((Allies)|(Axis))/(\d+)\)")
     log_time_regexp = re.compile(r".*\((\d+)\).*")
     connect_disconnect_pattern = re.compile(r"(.+) \((\d+)\)")
-    # Checking for steam ID length so people can't exploit it with a name like: short(Axis/123) ->
     kill_teamkill_pattern = re.compile(
-        r"(.*)\((?:Allies|Axis)\/(\d{17})\) -> (.*)\((?:Allies|Axis)\/(\d{17})\) with (.*)"
+        r"(.*)\((?:Allies|Axis)\/(.*)\) -> (.*)\((?:Allies|Axis)\/(.*)\) with (.*)"
```
Comment (FlorianSW) [inline rcon/rcon.py:111]: a bit greedy, we would probably be able to have discrete parts for either a numeric steam ID or a UUID (which seems to be the Windows Store ID). However, as we have no _guarantee_ that this is actually always a UUID, and given that the other possibilities of the whole line are expressed already, being greedy here is most likely no issue.

Scratch all what I wrote, I just was thinking out loud 😅

File: rcon/steam_utils.py [hunk 4]
```diff
@@ -15,6 +17,39 @@
 logger = logging.getLogger(__name__)
 
 last_steam_api_key_warning = datetime.datetime.now()
+steam_id_64_pattern = re.compile(r"\d{17}")
+
+
+def is_steam_id_64(steam_id_64: str) -> bool:
```
Comment (FlorianSW) [inline rcon/steam_utils.py:23]: Would probably be more performant if we do not use regular expressions for that. A Steam ID right now is always only digits and always 17 characters long. Something like this should be alright:

```
if steam_id_64.isdigit() and len(steam_id_64) == 17:
    return True
return False
```

File: rcon/steam_utils.py [hunk 5]
```diff
@@ -15,6 +17,39 @@
...
+    if re.match(steam_id_64_pattern, steam_id_64):
+        return True
+    return False
+
+
+def filter_steam_ids():
```
Comment (FlorianSW) [inline rcon/steam_utils.py:29]: I think for the time being, this is good enough for now. It abstracts away the explicit check if a player ID is a steam ID, however, for now that's fine.

I would wish that there will be _any_ way to get similar information about a player profile for a Windows Store version of HLL, but I doubt that we will get this in the near future or at all.

**Review by FlorianSW — CHANGES_REQUESTED** [review]
apart from my comments, this looks good :)

I didn't look for any place where we explicitly expect a Steam ID, however, I _think_ that this looks good so far :)

File: rcon/steam_utils.py (hunk 5 above)
Comment (cemathey) [reply rcon/steam_utils.py:29]: I've inquired but I doubt that we're ever going to have an API we can inquire against. It's not clear to me, but I believe these IDs are being generated by Team17 from some sort of windows store identifier :<

File: rcon/steam_utils.py (hunk 4 above)
Comment (cemathey) [reply rcon/steam_utils.py:23]: Much smarter

File: rcon/rcon.py (hunk 2 above)
Comment (cemathey) [reply rcon/rcon.py:49]: I'm partial to this name, it says exactly what it does.

I did update the docstring though which I forgot to do.

File: rcon/rcon.py (hunk 1 above)
Comment (cemathey) [reply rcon/rcon.py]: Updated.

**Review by ElGuillermo — APPROVED** [review]

File: rcon/steam_utils.py (hunk 5 above)
Comment (FlorianSW) [reply rcon/steam_utils.py:27]: > I believe these IDs are being generated by Team17 from some sort of windows store identifier

That's exactly what I got from the document shared by T17 as well 🙈

**Review by FlorianSW — APPROVED** [review]

**Pull request merged**